firestore-counter/clients/ios/Sources/FirestoreCounter/FirestoreCounter.swift

```diff
@@ -0,0 +1,91 @@
+import FirebaseFirestore
+import Firebase
+
+public class FirestoreShardCounter {
+  private let firestore: Firestore
+  private let shardId = UUID().uuidString
+  private var shards = [String: Double]()
+  private let collectionId = "_counter_shards_"
+  private let shardRef: CollectionReference
+  private var docRef: DocumentReference
+  private var field: String
+
+  public init(docRef: DocumentReference, field: String) {
+    self.docRef = docRef
+    self.field = field
+    self.firestore = docRef.firestore
+    self.shardRef = docRef.collection(collectionId)
+    shards[self.docRef.path] = Double(0)
+    shards[self.shardRef.document(String(self.shardId)).path] = Double(0)
+    shards[self.shardRef.document("\t" + String(self.shardId.prefix(4))).path] = Double(0)
+    shards[self.shardRef.document("\t\t" + String(self.shardId.prefix(3))).path] = Double(0)
+    shards[self.shardRef.document("\t\t\t" + String(self.shardId.prefix(2))).path] = Double(0)
+    shards[self.shardRef.document("\t\t\t\t" + String(self.shardId.prefix(1))).path] = Double(0)
+  }
+
+  public func get(result: @escaping (Double?, Error?) -> ()) {
+    var documentIds = [String]()
+    var totalValue = Double(0)
+
+    for key in shards.keys {
+      documentIds.append((key as NSString).lastPathComponent)
+    }
+
+    shardRef.whereField(FieldPath.documentID(), in: documentIds).getDocuments() { (snapshot, error) in
```

Comment:
Style guide on the main iOS SDK is to remove the parens around the closure parameter list, i.e.
```
hardRef.whereField(FieldPath.documentID(), in: documentIds).getDocuments() { snapshot, error in
```